### PR TITLE
Better find in documentation

### LINKED ISSUE #14410: Better find in documentation
The current find function in the documentation is not convenient. It's a popup, it hides part of the doc, and the doc is hard to read because it's darkened, there's no Find next/Find previous shortcut.

If possible the find function should be the same as when editing scripts, which has a much better design.

### PR DESCRIPTION
**Pull request by guilhermefelipecgs**
#### [ Reopening this PR because I deleted my branch unintentionally. ref. #14950 ]

Fix #14410, #14925 

I've changed "search dialog" in the documentation for "search bar", as in the script editor, with options for searching next and previous text.

2017-12-22-164704_1113x374_scrot

### Update:
2018-04-10-131550_958x128_scrot

### CODE REVIEW

**Comment by akien-mga** [comment]
We are now entering a strict release freeze for Godot 3.0 and will only consider major bug fixes. We won't merge new features and enhancements anymore until 3.0 is released.

Moving this PR to the 3.1 milestone, to be reviewed once the release freeze is lifted. It could eventually be cherry-picked for a future 3.0.1 maintenance release if it doesn't change the user-facing APIs and doesn't compromise the engine's stability.

**Comment by guilhermefelipecgs** [comment]
Can we "cherrypick" this pr? I made a rebase and tested again and I think it's safe to merge.

**Comment by akien-mga** [comment]
It needs to be merged and tested in master first, but then yeah it could be cherry-picked.

**Review by toger5 — COMMENTED** [review]
Looks good to me

File: editor/editor_help.cpp [hunk 1]
```diff
@@ -1964,3 +1932,178 @@ EditorHelpBit::EditorHelpBit() {
...
+	HBoxContainer *search_options = memnew(HBoxContainer);
+	options_vbc->add_child(search_options);
+
+	error_label = memnew(Label);
+	search_options->add_child(error_label);
+	error_label->add_color_override("font_color", EditorNode::get_singleton()->get_gui_base()->get_color("error_color", "Editor"));
+
+	search_options->add_spacer();
+
+	hide_button = memnew(TextureButton);
```
[the remark this answers is not in the document]
Comment (guilhermefelipecgs) [reply editor/editor_help.cpp]: You're right, I'll fix this.

**Comment by toger5** [comment]
Im suuper thankful you changed the close icon.
I also love the wider search bar!!

The rest of the code also looks good to me.


Maybe the search bar in the code editor could be adapted to also have a wider text field?

**Comment by reduz** [comment]
What's the status of this?

**Comment by guilhermefelipecgs** [comment]
It's done.

**Comment by akien-mga** [comment]
Thanks!

**Pull request merged**